src/main/java/org/opensearch/plugin/insights/rules/model/SearchQueryRecord.java

```diff
@@ -259,8 +262,22 @@ public static SearchQueryRecord fromXContent(XContentParser parser) throws IOExc
                         attributes.put(Attribute.QUERY_GROUP_HASHCODE, parser.text());
                         break;
                     case SOURCE:
-                        XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
-                        attributes.put(Attribute.SOURCE, SearchSourceBuilder.fromXContent(parser, false));
+                        if (parser.currentToken() == XContentParser.Token.VALUE_STRING) {
+                            String sourceString = parser.text();
+                            try {
+                                XContentParser sourceParser = JsonXContent.jsonXContent.createParser(
+                                    NamedXContentRegistry.EMPTY,
+                                    DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+                                    sourceString
+                                );
+                                attributes.put(Attribute.SOURCE, SearchSourceBuilder.fromXContent(sourceParser, false));
+                            } catch (Exception e) {
+                                attributes.put(Attribute.SOURCE, sourceString);
+                            }
+                        } else {
+                            XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+                            attributes.put(Attribute.SOURCE, SearchSourceBuilder.fromXContent(parser, false));
```

Comment:
I thought this PR is converting the in-memory representation of source to `String`. What scenario do we need `SearchSourceBuilder.fromXContent`?